packages/muelu/src/Graph/MatrixTransformation/MueLu_CoalesceDropFactory_def.hpp

```diff
@@ -572,109 +577,172 @@ void CoalesceDropFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Build(Level
               }
               rows(row + 1) = realnnz;
             }
-          } else {
-            /* Cut Algorithm */
-            // CMS
-            using DropTol = Details::DropTol<real_type, LO>;
-            std::vector<DropTol> drop_vec;
-            drop_vec.reserve(nnz);
-            const real_type zero = Teuchos::ScalarTraits<real_type>::zero();
-            const real_type one  = Teuchos::ScalarTraits<real_type>::one();
-            LO rownnz            = 0;
-            // NOTE: This probably needs to be fixed for rowsum
-
-            // find magnitudes
-            for (LO colID = 0; colID < (LO)nnz; colID++) {
-              LO col = indices[colID];
-              if (row == col) {
-                drop_vec.emplace_back(zero, one, colID, false);
-                continue;
+          }  // end for row
+        }
+        else {
+          /* Cut Algorithm */
+          SubFactoryMonitor m1(*this, "Cut Drop", currentLevel);
+          using ExecSpace = typename Node::execution_space;
+          using TeamPol = Kokkos::TeamPolicy<ExecSpace>;
+          using TeamMem = typename TeamPol::member_type;
+          using ATS = Kokkos::ArithTraits<Scalar>;
+          using impl_scalar_type = typename ATS::val_type;
+          using implATS = Kokkos::ArithTraits<impl_scalar_type>;
+
+          //move from host to device
+          auto ghostedDiagValsView = Kokkos::subview(ghostedDiag->getDeviceLocalView(Xpetra::Access::ReadOnly), Kokkos::ALL(), 0);
+          auto boundaryNodesDevice = Kokkos::create_mirror_view(ExecSpace(), boundaryNodes);
+          auto thresholdKokkos = static_cast<impl_scalar_type>(threshold);
+          auto realThresholdKokkos = implATS::magnitude(thresholdKokkos);
+
+          auto At = Utilities::Op2TpetraCrs(A);
+          auto A_device = At->getLocalMatrixDevice();
+
+          int algorithm = classicalAlgo;
```

Comment:
Do we need to do this? We already have variables `algo` and `classicalAlgo` floating around.

Reply:
classicalAlgo can't be used on device because "A type local to a function ("decisionAlgoType") cannot be used in the type of a variable captured by an extended __device__ or __host__ __device__ lambda".
algo can't be used because this variable is meant to determine whether a classical scaled cut is being executed or a classical unscaled cut is being executed. In either of these scenarios, algo is set to the same string "classical".